### PR TITLE
Add Wii Platform Clock

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #2186

### CODE REVIEW

**Comment by Ghabry** [comment]
In Dolphin this works fine but maybe also do a hardware test. Don't have my Wii around so maybe @carstene1ns could do it?

**Comment by Ghabry** [comment]
When I do ``std::chrono::time_point_cast<std::chrono::milliseconds>(Platform_Clock::now())`` I get on Linux: 9075040, 9076040 and in Dolphin: 
Some gigantic number such as "5021779348504474546" followed by "5283566805015642470"

**Comment by Ghabry** [comment]
Test code for the clock.
Just put it in the main before ``Player::Init`` is called

```cpp
#include "platform_clock.h"
#include "output.h"

	Output::Debug("%s", Platform_Clock::Name());
	auto a = Platform_Clock::now();
	auto b = Platform_Clock::now();

	do {
		a = Platform_Clock::now();
		Output::Debug("Now1: %lld %lld", a - b, std::chrono::duration_cast<std::chrono::milliseconds>(a - b));
		Output::Debug("Sleep 1s");
		Platform_Clock::SleepFor(1s);
		b = Platform_Clock::now();
		Output::Debug("Now2: %lld %lld", b - a, std::chrono::duration_cast<std::chrono::milliseconds>(b - a));
		Output::Debug("Sleep 1s");
		Platform_Clock::SleepFor(1s);
	} while (true);

**Comment by Ghabry** [comment]
I'm so sorry for wasting time here. Was all my fault.

The correct way to print a chrono time is
```
std::chrono::duration_cast<std::chrono::milliseconds>(b - a).count()
```
and not
```
std::chrono::duration_cast<std::chrono::milliseconds>(b - a)
```

Somehow this worked on Linux and 3DS so I thought this is the correct way to do it :/

Even the first commit after fixing the 2 compile errors yields 1000.

**Comment by Ghabry** [comment]
Guess the reason why this printed wrong without the count() call is that the way powerpc packs 64bit different in memory (different byte order, also cpu is not 64bit) :(

Wii: ``39f8cc40000940d``
Linux: ``3b9c9635``

File: src/std_clock.h [hunk 1]
```diff
@@ -19,15 +19,11 @@
...
-#include <type_traits>
 #include <thread>
 
 #if !defined(_3DS) && !defined(__SWITCH__) && !defined(PSP2)
 struct StdClock {
-	using clock = std::conditional_t<
-		std::chrono::high_resolution_clock::is_steady,
-		std::chrono::high_resolution_clock,
-		std::chrono::steady_clock>;
+	using clock = std::chrono::steady_clock;
```
Comment (carstene1ns) [inline src/std_clock.h:26]: Looks like the logic about high res clock can be removed in `Name()` function as well?

**Comment by Ghabry** [comment]
btw Android actually reports highres clock but I assume this is an artifact of the evaluation order. When the check for HighRes clock in Name() would be the last this would report a different clock.

```
Clock: StdHigRes steady=1 period=ns (1 / 1000000000)
```

**Comment by carstene1ns** [comment]
Makes sense, so it is both HighRes and Steady.

**Comment by carstene1ns** [comment]
I can confirm that this works on hardware.
@fmatthew5876: Question is, do you still want to revert some part of it (as @Ghabry said it worked already in a former state).

Once merged, I will pull this into maintenance branch.

**Comment by Ghabry** [comment]
This is already the first version.
He reverted it for me once I found the problem in my tests. :)

**Comment by carstene1ns** [comment]
Okay, then ready to merge after having heard about the `StdClock::Name()`.

**Comment by fmatthew5876** [comment]
Re-ordered the name check. I kept the ifs in there just in case later we decide we need to use some of the other standard clocks. They are compile time ifs so performance is not affected.

**Comment by fmatthew5876** [comment]
jenkins: test this please

**Pull request merged**